### PR TITLE
Ria 3309 ho hearing

### PR DESCRIPTION
**Pull request by Lee365**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-3309

### Change description ###
RIA-3309 ho hearing

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/functionalTest/resources/scenarios/RIA-3309-notification-for-hearing-edit-case-listing-decision.json [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        "listCaseHearingLength": "60",
+        "listCaseHearingDate": "2018-12-31T12:34:56",
+        "listCaseRequirementsVulnerabilities": "something around Vulnerabilities",
+        "listCaseRequirementsMultimedia": "something around Multimedia",
+        "listCaseRequirementsSingleSexCourt": "something around SingleSexCourt",
+        "listCaseRequirementsInCameraCourt": "something around InCameraCourt",
+        "listCaseRequirementsOther": "something around Other",
+        "uploadAdditionalEvidenceActionAvailable": "Yes",
+        "ariaListingReference": "LP/12345/2019",
+        "homeOfficeHearingInstructStatus": "OK"
```
Comment (jkumar-moj) [inline src/functionalTest/resources/scenarios/RIA-3309-notification-for-hearing-edit-case-listing-decision.json:66]: I mentioned earlier about having a separare flag. do you think it is not needed? The same comment applies to all tests and handler

File: src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/ListCaseNotificationHandlerTest.java [hunk 2]
```diff
@@ -70,11 +73,12 @@ void setUp() {
                 dateTimeExtractor);
     }
 
-    @Test
+    @ParameterizedTest
```
Comment (jkumar-moj) [inline src/test/java/uk/gov/hmcts/reform/iahomeofficeintegrationapi/domain/handlers/presubmit/ListCaseNotificationHandlerTest.java:76]: good one..  👍

**Review by jkumar-moj — CHANGES_REQUESTED** [review]

File: src/functionalTest/resources/scenarios/RIA-3309-notification-for-hearing-edit-case-listing-decision.json (hunk 1 above)
Comment (Lee365) [reply src/functionalTest/resources/scenarios/RIA-3309-notification-for-hearing-edit-case-listing-decision.json:66]: No, I don't think there is a need for separate flag.

File: src/functionalTest/resources/scenarios/RIA-3309-notification-for-hearing-edit-case-listing-prehearing.json [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        "listCaseHearingLength": "60",
+        "listCaseHearingDate": "2018-12-31T12:34:56",
+        "listCaseRequirementsVulnerabilities": "something around Vulnerabilities",
+        "listCaseRequirementsMultimedia": "something around Multimedia",
+        "listCaseRequirementsSingleSexCourt": "something around SingleSexCourt",
+        "listCaseRequirementsInCameraCourt": "something around InCameraCourt",
+        "listCaseRequirementsOther": "something around Other",
+        "uploadAdditionalEvidenceActionAvailable": "Yes",
+        "ariaListingReference": "LP/12345/2019",
+        "homeOfficeHearingInstructStatus": "OK"
```
[the remark this answers is not in the document]
Comment (Lee365) [reply src/functionalTest/resources/scenarios/RIA-3309-notification-for-hearing-edit-case-listing-prehearing.json:66]: No, I don't think there is a need for separate flag.

File: src/functionalTest/resources/scenarios/RIA-3309-notification-for-hearing-edit-case-listing-prehearing.json [hunk 4]
```diff
@@ -0,0 +1,70 @@
+{
+  "description": "RIA-3309 Sends edit case listing notification to Home office API for state prehearing",
+  "disabled": false,
+  "request": {
+    "uri": "/asylum/ccdAboutToSubmit",
+    "credentials": "AdminOfficer",
```
[the remark this answers is not in the document]
Comment (Lee365) [reply src/functionalTest/resources/scenarios/RIA-3309-notification-for-hearing-edit-case-listing-prehearing.json:6]: Test added to show editCaseListing done by case officer

File: src/functionalTest/resources/scenarios/RIA-3309-notification-for-hearing-edit-case-listing-decision.json (hunk 1 above)
Comment (jkumar-moj) [reply src/functionalTest/resources/scenarios/RIA-3309-notification-for-hearing-edit-case-listing-decision.json:66]: But then there is no way of knowing if there were any edit on listing and what is the notification status for that isn't it? I think it is good to have a status flag. Also as I mentioned earlier if we have to display a message to TCW we need a flag to differentiate as well. Let me know if you still think it is not needed and we can discuss

**Review by jkumar-moj — CHANGES_REQUESTED** [review]

**Review by jkumar-moj — APPROVED** [review]
looks good. 👍

File: src/functionalTest/resources/scenarios/RIA-3309-notification-for-hearing-edit-case-listing-decision.json (hunk 1 above)
Comment (Lee365) [reply src/functionalTest/resources/scenarios/RIA-3309-notification-for-hearing-edit-case-listing-decision.json]: New status homeOfficeEditListingInstructStatus created

**Review by Srihari-Nakkala — APPROVED** [review]
LGTM

**Comment by jkumar-moj** [comment]
Re-opening the PR as it was closed by stalebot

**Pull request merged**